mux/tree.go

```diff
@@ -134,8 +134,12 @@ func (t Tree) WithMiddleware(path string, m middleware.Middleware, maxParamsSize
 	node := t.Find(name)
 	newTree := t
 
+	// If there is no node get the route of the previous pameterized one
+	// and append middleware on top of its own middleware
 	if node == nil {
 		node = NewNode(parts[0], maxParamsSize)
+		node.WithRoute(t.Find("{}").Route())
```

Comment:
what is this ? `t.Find("{}")`

Reply:
To be honest I am not sure that this issue, can be solved with the current tree implementation.
I believe that this **if** statement should panic instead of creating new route. This is due to following reason.

If we have a route for a wildcard path, but add middleware for a static path (that should be matched by wildcard one). Because we create a tree node now, without a route, while performing request match it will match the static node without route and throw 404. Therefore I believe we should `panic` here and leave it up to the user to implement his handler the way he needs it to be executed. He could even use subrouter here.

Reply:
In my opinion a developer expects by default a middleware be mapped to a route out of the box. Since in the current tree implementation the adding of routes happens in `route.GET` during tree generation by calling `withRoute` it seems logical that the next step when calling withMiddleware(essentially appending middleware to the already generated tree) would be an attempt to map this middleware predefined static route to the first matching wildcard one moving backwards up the tree, otherwise it can be considered an orphaned one.

Reply:
Maybe then instead of only finding node by its name, we should try to find a matching one if by name can not be found ?

Reply:
Yes exactly that's even a better place than t.Find() because it returns the middleware we can append directly to.